commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java

```diff
@@ -37,9 +37,19 @@ public Logger getLog() {
 	return log;
     }
 
-    protected RsaVerifier getSignatureVerifier() {
-	if (signatureVerifier == null)
-	    signatureVerifier = new RsaVerifier(getSignatureKey());
+    protected RsaVerifier getSignatureVerifier() throws ApplicationAuthenticationException {
+	if (signatureVerifier == null) {
+		String signatureKey = getSignatureKey();
+	    if(signatureKey==null || signatureKey.isBlank()) {

```

Comment:
This is a configuration error, so it should be handled within each application – not in api-commons.

Ideally an error should be thrown on startup if no signature key is configured.

Reply:
Done.